### ISSUE TITLE
Webhook doesn't fire.

### ISSUE THREAD

**Issue by onebelarusianguy** [stripe/stripe-node#1095]
Hi, I can't get working webhooks. I'm sending a POST request to '/create-session' route, filing payment details after redirect and expecting to see any event (or `⚠️  Webhook call` for this case) fire in the **webhook** function, but it doesn't.

### NodeJS example to reproduce:

```
const MODE = 'test';
const STATIC_DIR = '../public';
const express = require('express');
const { resolve } = require('path');
const app = express();
const router = express.Router();
const appConfig = require(`./config-app-${MODE}.json`);
const stripe = require('stripe')(appConfig.stripeSecret);
app.use(express.static(STATIC_DIR));


app.get('/', (req, res) => {
  const path = resolve(STATIC_DIR + '/index.html');
  res.sendFile(path);
});


app.post('/create-session', async (req, res) => {
    const session = await stripe.checkout.sessions.create({
        payment_method_types: ['card'],
        line_items: [{
            price_data: {
                currency: 'usd',
                product_data: {
                    name: 'T-shirt',
                },
                unit_amount: 2000,
            },
            quantity: 1,
        }, ],
        mode: 'payment',
        success_url: `http://localhost:4242/success?session_id={CHECKOUT_SESSION_ID}`,
        cancel_url: 'http://localhost:4242/cancel'
    });

    res.json({
        id: session.id
    });
});


async function webhook(req, res) {
    console.log(`⚠️  Webhook call`);

    res.sendStatus(200);
}


function send(res, code, content) {
    res.send({
        statusCode: code,
        body: JSON.stringify(content)
    });
}

function successRoute(req, res) {
    send(res, 200, {
        success: true
    });
}

function cancelRoute(req, res) {
    send(res, 200, {
        success: false
    });
}

app.post('/webhook', webhook);
app.get('/success', successRoute);
app.get('/cancel', cancelRoute);


app.listen(4242, () => console.log(`Listening on port ${4242}!`));
```
and 

`stripe listen --forward-to localhost:4242/webhook`

### Expectations:
<img width="437" alt="Screen Shot 2020-12-23 at 03 08 44" src="https://user-images.githubusercontent.com/13021107/102944694-93493e00-44cc-11eb-94ed-45094fd50397.png">
<img width="791" alt="Screen Shot 2020-12-23 at 03 08 12" src="https://user-images.githubusercontent.com/13021107/102944688-8f1d2080-44cc-11eb-9e48-0d6658da809d.png">

### Reality:
nothing.

**The webhook endpoint only fires when I send** `Send a test event to a webhook endpoint` **form Stripe Dashboard
or
when I use Stripe-cli:** `stripe trigger {EVENT_TYPE}`

_Node version: from 10 to 12
Stripe version: 8.129.0_

I also tried [https://github.com/stripe-samples/checkout-one-time-payments](url) nodejs example. The same stuff.

What's wrong?

**Comment by paulasjes-stripe** [comment]
When you say "nothing" do you mean you aren't getting the `console.log` calls? Are you able to successfully create Checkout Sessions? Any errors in your console?

**Comment by onebelarusianguy** [comment]
@paulasjes-stripe 

Yes, no console logs - the function is not called at all. And stripe listener doesn’t show any post requests to /webhook.
Session is created and return its id. There is no errors. Once again: webhook function is called when I trigger event with stripe dashboard or using stripe-cli. But it’s not when I create a session (or payment intention for example) by code. I would never ask a question if there were errors or something obvious. I understand what I am doing. I didn’t asked questions regarding programming about 4 years. This one is really weird.

**Comment by paulasjes-stripe** [comment]
You should be able to see failed CLI webhook attempts in your dashboard: https://dashboard.stripe.com/test/webhooks

Can you check there to see if there are any failed `checkout.session.completed` events?

**Comment by onebelarusianguy** [comment]
@paulasjes-stripe There is nothing in dashboard. I captured a [video](https://youtu.be/tKYIfhOJYSw) using server-side code I provided here.

Events appears only when I trigger it with stripe-cli.
And actually I do not understand why when I trigger it from my localhost they appears in 
https://test-73c6a.web.app/webhook webhook, why? This webhook I created for production.
There should not be a webhook in the dashboard since I use stripe-cli and can't add localhost as a domain, right?
